cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -0,0 +1,207 @@
+}
+
+// Scheduler specifies how the scheduler component is set up and configured
+type Scheduler struct {
+	// Image specifies which container image to use for running the scheduler. If empty, automatically populated by kubeadm using the ImageRepository and KubernetesVersion
+	Image string `json:"image"`
+	// ExtraArgs lets the user customize and/or override the arguments that are given to the scheduler
+	ExtraArgs map[string]string `json:"extraArgs"`
+	// ExtraHostPathVolumes lets the user add new hostPath volumes and volumeMounts as needed
+	ExtraHostPathVolumes []HostPathVolume `json:"extraHostPathVolumes"`
+}
+
+// HostPathVolume is a high-level abstraction for a v1.Volume and v1.VolumeMount attached to the Pod the component is running in
+type HostPathVolume struct {
+	// Name specifies the name of the Volume and VolumeMount
+	Name string `json:"name"`
+	// HostPath specifies the path on the host (can be a file or a directory) that should be mounted into the container
+	HostPath string `json:"hostPath"`
+	// ContainerPath specifies the path inside of the container (can be a file or a directory) that should contain the data mounted from the host
+	ContainerPath string `json:"containerPath"`
+}
+
+// Networking defines a couple of cluster-wide properties related to the networking setup
+type Networking struct {
+	// ServiceSubnet specifies which subnet to use for virtual Service IPs
+	ServiceSubnet string `json:"serviceSubnet"`
+	// PodSubnet specifies which subnet to use for Pods in the cluster. Setting this is optional.
+	// Setting this is enables the controller-manager's built-in subnet allocator that will set Node.Spec.PodCIDR,
+	// which is required information for some third-party CNI network plugins
+	PodSubnet string `json:"podSubnet"`
+	// DNSDomain sets the DNS domain for the cluster; currently used when generating the API Server serving certificate
+	DNSDomain string `json:"dnsDomain"`
+}
+
+// BootstrapToken controls how/if bootstrap tokens should be enabled or how they should be created
+type BootstrapToken struct {
+	// Node describes a Node Bootstrap Token that is automatically generated on cluster initialization
+	// If this pointer is nil, this field defaults to being enabled, having an auto-generated token and having a time-limited ttl 
+	Node *NodeBootstrapToken `json:"node"`
+}
+
+// NodeBootstrapToken controls how/if node bootstrap tokens should be enabled or how it should be created
+type NodeBootstrapToken struct {
+	// Enabled specifies whether a Node Bootstrap Token should automatically be generated and the cluster-info ConfigMap should be created
+	Enabled bool `json:"enabled"`
+	// Token specifies which Node Bootstrap Token in the "<6 chars>.<16 chars>" format to automatically add to the cluster
+	Token string `json:"token"`
+	// TokenTTL specifies how long the automatically generated Node Bootstrap Token should be valid for
+	TokenTTL time.Duration `json:"tokenTTL"`
```

Comment:
this should be metav1.Duration, shouldn't it? A raw duration will not marshal correctly.